docs/tutorials/start-a-private-network-with-substrate.md

```diff
@@ -9,29 +9,33 @@ You will need:
 
 You will learn to:
 1. Compile a Substrate node from source code.
-2. Generate ed25519 key-pairs for use as a network authority
+2. Generate ed25519 and sr25519 key-pairs for use as a network authority
 3. Create and edit a chainspec json file
 
 
 Compiling the Tools
 -----------------
-Substrate is an open source framework for building your own blockchain nodes. It can be used for a very wide variety of applications.
+Substrate does not (yet) offer binary installation packages, so it must be compiled from source, which can be a time-consuming process. Commits go into the [Substrate repository](https://github.com/paritytech/substrate) regularly and it is wise that everyone participating in this network have the same version of substrate to guarantee success. In practice similar but not identical versions will often work, but banking on that is a recipe for frustration. In this tutorial, we'll be using commit `7d7e74fb` for more stability.
 
-Substrate does not (yet) offer binary installation packages, so it must be compiled from source, which can be a time-consuming process. Commits go into the [Substrate repository](https://github.com/paritytech/substrate) regularly and it is wise that everyone participating in this network have the same version of substrate to guarantee success. In practice similar but not identical versions will often work, but banking on that is a recipe for frustration. We'll be using the `v1.0` branch for more stability.
-
-### Your First Time
-First we'll download the Substrate v1.0 code by cloning the git repository.
 ```bash
-git clone -b v1.0 https://github.com/paritytech/substrate
+# Download the Substrate code
+git clone https://github.com/paritytech/substrate
 cd substrate
+
+# Switch to the proper commit
+git checkout 7d7e74fb
+
+# Install rust prerequisites
+curl https://getsubstrate.io -sSf | bash -s -- --fast
```

Comment:
I'd like to put this step in the first place, since it may take a long time.

Reply:
I'd also like to see more about what this is downloading / doing, without having to go to the script myself. This is important since we're piping into bash, which is generally a security concern.

Reply:
I totally agree about. The reason I didn't write that kind of stuff here is because I believe the plan is to have some unified install instructions that I should just link to from here.

Reply:
I agree. I didn't write that material here because I hope we can have setup-your-developer-environment instructions on a dedicated page. That's my second checkbox in the description.